## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#fa7ac6b6-da6a-4274-9239-3de5aee6e606) |
<a name="fa7ac6b6-da6a-4274-9239-3de5aee6e606"></a>
### 10917-c47d9a2/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev8.minio.io:30727
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 43667c45d4c3:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 9 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 2 minutes and 15 seconds
(4/15) Running aws-sdk-ruby tests ... done in 20 seconds
(5/15) Running awscli tests ... done in 3 minutes and 19 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 3 minutes and 42 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 42 seconds
(9/15) Running minio-go tests ... done in 6 minutes and 23 seconds
(10/15) Running minio-java tests ... done in 11 minutes and 4 seconds
(11/15) Running minio-js tests ... done in 2 minutes and 39 seconds
(12/15) Running minio-py tests ... done in 19 minutes and 54 seconds
(13/15) Running s3cmd tests ... FAILED in 1 minutes and 20 seconds
{
  "name": "s3cmd",
  "duration": "18340",
  "function": "test_get_object_multipart",
  "status": "FAIL",
  "error": "\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions\n  If the error persists, please report the\n  following lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n\nInvoked as: /usr/local/bin/s3cmd --config /tmp/.s3cmd-20457/s3cfg get s3://s3cmd-test-bucket-25875/s3cmd-test-object-29955 s3cmd-test-object-29955.downloaded\nProblem: <class 'ConnectionResetError: [Errno 104] Connection reset by peer\nS3cmd:   2.1.0\npython:   3.6.9 (default, Jul 17 2020, 12:50:27) \n[GCC 8.4.0]\nenvironment LANG=C.UTF-8\n\nTraceback (most recent call last):\n  File \"/usr/local/bin/s3cmd\", line 3121, in <module>\n    rc = main()\n  File \"/usr/local/bin/s3cmd\", line 3030, in main\n    rc = cmd_func(args)\n  File \"/usr/local/bin/s3cmd\", line 604, in cmd_object_get\n    response = s3.object_get(uri, dst_stream, destination, start_position = start_position, extra_label = seq_label)\n  File \"/usr/local/lib/python3.6/dist-packages/S3/S3.py\", line 726, in object_get\n    response = self.recv_file(request, stream, labels, start_position)\n  File \"/usr/local/lib/python3.6/dist-packages/S3/S3.py\", line 1733, in recv_file\n    data = http_response.read(this_chunk)\n  File \"/usr/lib/python3.6/http/client.py\", line 459, in read\n    n = self.readinto(b)\n  File \"/usr/lib/python3.6/http/client.py\", line 503, in readinto\n    n = self.fp.readinto(b)\n  File \"/usr/lib/python3.6/socket.py\", line 586, in readinto\n    return self._sock.recv_into(b)\nConnectionResetError: [Errno 104] Connection reset by peer\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions\n  If the error persists, please report the\n  above lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
}
(13/15) Running s3select tests ... done in 1 minutes and 11 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/metacache-set.go
+++ b/cmd/metacache-set.go
@@ -21,6 +21,7 @@ import (
 	"context"
 	"encoding/gob"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"io"
 	"strconv"
@@ -343,6 +344,7 @@ func (r *metacacheReader) filter(o listPathOptions) (entries metaCacheEntriesSor
 func (er *erasureObjects) streamMetadataParts(ctx context.Context, o listPathOptions) (entries metaCacheEntriesSorted, err error) {
 	retries := 0
 	rpc := globalNotificationSys.restClientFromHash(o.Bucket)
+
 	for {
 		select {
 		case <-ctx.Done():
@@ -395,6 +397,7 @@ func (er *erasureObjects) streamMetadataParts(ctx context.Context, o listPathOpt
 				return entries, fmt.Errorf("reading first part metadata: %w", err)
 			}
 		}
+
 		if fi.Deleted {
 			return entries, errFileNotFound
 		}
@@ -531,14 +534,24 @@ func (er *erasureObjects) listPath(ctx context.Context, o listPathOptions) (entr
 	if debugPrint {
 		console.Printf("listPath with options: %#v\n", o)
 	}
+
 	// See if we have the listing stored.
 	if !o.Create && !o.singleObject {
 		entries, err := er.streamMetadataParts(ctx, o)
-		switch err {
-		case nil, io.EOF, context.Canceled, context.DeadlineExceeded:
-			return entries, err
+		if IsErr(err, []error{
+			nil,
+			context.Canceled,
+			context.DeadlineExceeded,
+		}...) {
+			// Expected good errors we don't need to return error.
+			return entries, nil
 		}
-		logger.LogIf(ctx, err)
+
+		if !errors.Is(err, io.EOF) { // io.EOF is expected and should be returned but no need to log it.
+			// Log an return errors on unexpected errors.
+			logger.LogIf(ctx, err)
+		}
+
 		return entries, err
 	}
 
@@ -559,7 +572,7 @@ func (er *erasureObjects) listPath(ctx context.Context, o listPathOptions) (entr
 		if debugPrint {
 			console.Println("listPath returning:", entries.len(), "err:", err)
 		}
-		if err != nil && err != io.EOF {
+		if err != nil && !errors.Is(err, io.EOF) {
 			go func(err string) {
 				metaMu.Lock()
 				if meta.status != scanStateError {

